pkg/awsutils/awsutils.go

```diff
@@ -315,8 +331,27 @@ func (i instrumentedIMDS) GetMetadataWithContext(ctx context.Context, p string)
 	return result, nil
 }
 
+func (cache *EC2InstanceMetadataCache) GetNetworkInterfacesWithContext(ctx context.Context, eniIds []*string) (*ec2.DescribeNetworkInterfacesOutput, error) {
+	input := &ec2.DescribeNetworkInterfacesInput{NetworkInterfaceIds: eniIds}
+
+	start := time.Now()
```

Comment:
not related to this PR(since all other functions are already collecting metrics this way).
but for aws api related metrics, it's better to collect it this way: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/aws/metrics/collector.go
which is not intrusive to application code.

Reply:
Yeah makes sense, will add a todo and take up as part of clean up efforts.